realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -138,6 +143,279 @@ public boolean load() {
+    }
+
+    /**
+     * Sets the {@code String} value of the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't string.
+     * @throws RealmException if the field is a {@link io.realm.annotations.PrimaryKey} field.
+     */
+    public void setString(@Nonnull String fieldName, @Nullable String value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        collection.setString(getColumnIndex(fieldName), value);
+    }
+
+    /**
+     * Sets the binary value of the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't binary.
+     */
+    public void setBlob(@Nonnull String fieldName, @Nullable byte[] value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        collection.setBinary(getColumnIndex(fieldName), value);
+    }
+
+    /**
+     * Sets the {@code Date} value of the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't date.
+     */
+    public void setDate(@Nonnull String fieldName, @Nullable Date value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        collection.setDate(getColumnIndex(fieldName), value.getTime());
+    }
+
+    /**
+     * Sets a reference to another object on the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or is of the wrong type.
+     */
+    public void setObject(@Nonnull String fieldName, @Nullable RealmObject value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        long columnIndex = getColumnIndex(fieldName);
+
+        if (!(value instanceof RealmObjectProxy)) {
+            throw new IllegalArgumentException("A field value must be a managed Realm object.");
+        }
+
+        ProxyState proxyState = ((RealmObjectProxy) value).realmGet$proxyState();
+        if (proxyState.getRealm$realm() != realm) {
+            throw new IllegalArgumentException("A field value must belong to the same Realm as its owner.");
+        }
+
```

Comment:
Well, in this case you are talking about the input parameter, of which there is only one, not the collection right?